skfp/model_selection/utils.py

```diff
@@ -0,0 +1,87 @@
+import warnings
+from collections.abc import Sequence
+from itertools import chain
+from typing import Any, Optional, Union
+
+import numpy as np
+from rdkit.Chem import Mol
+from sklearn.utils import _safe_indexing
+
+
+def ensure_nonempty_list(subset: list) -> None:
+    """
+    Check if the provided subset is empty.
+    """
+    if len(subset) == 0:
+        raise ValueError("One of the subsets is empty.")
+
+
+def fill_missing_sizes(
+    train_size: Optional[float], test_size: Optional[float]
+) -> tuple[float, float]:
+    """
+    Fill in missing sizes for train and test sets based on the provided sizes.
+    """
+    if train_size is None and test_size is None:
+        train_size = 0.8
+        test_size = 0.2
+    if train_size is None:
+        if test_size is not None:
+            train_size = 1 - test_size
+        else:
+            raise ValueError("test_size must be provided when train_size is None")
```

Comment:
This is not true, first `if` covers this case. After `if train_size is None and test_size is None`, you know that at least one of those is not `None`.

Reply:
So this part simplifies to:
```
if train_size is None:
    test_size = 1 - train_size
elif test_size is None:
    train_size = 1 - test_size
else:
    # check that they sum to approximately 1.0
    ...

return train_size, test_size
```

The default case is that both are provided and not `None`, so we can check here if they properly sum up to 1.0